### ISSUE TITLE
SQLite truncates strings with \u0000 on iOS

### ISSUE THREAD

**Issue by martin-ecatch** [expo/expo#5438]
## 🐛 Bug Report
SQLite truncates strings when character \u0000 is used. Everything after this character is lost.
Why does it matter? PouchDB with SQLite adapter routinely uses \u0000 in mapreduce views (very important part of PouchDB). Views don't work because of this. A workaround is to use asyncstorage adapter instead, but that is orders of magnitude slower (and therefore pretty much unusable).
This issue is solved for Android (see _serializeQuery and _escapeBlob functions in expo-sqlite), but breaks on iOS.

### Environment
  Expo CLI 3.0.9 environment info:
    System:
      OS: Linux 4.15 Ubuntu 18.04.3 LTS (Bionic Beaver)
      Shell: 4.4.20 - /bin/bash
    Binaries:
      Node: 10.16.3 - /usr/bin/node
      Yarn: 1.17.3 - /usr/bin/yarn
      npm: 6.9.0 - /usr/bin/npm
      Watchman: 4.9.0 - /usr/local/bin/watchman
    npmPackages:
      expo: ^33.0.0 => 33.0.7 
      react: 16.8.3 => 16.8.3 
      react-native: https://github.com/expo/react-native/archive/sdk-33.0.0.tar.gz => 0.59.8 
    npmGlobalPackages:
      expo-cli: 3.0.9

This happens on iOS in both Expo Client and standalone apps.

### Steps to Reproduce
Snack here https://snack.expo.io/@ecatch/test-sqlite
Add a new item to the TODO list (name it "Something"). See on line 118 it says:
`    text = text + '\u0000 this will be truncated';`

### Expected Behavior
I'd expect the value "Something\u0000 this will be truncated" to be stored in SQLite.

### Actual Behavior
Value "Something" is stored in SQLite (truncation happens upon \u0000).

### Reproducible Demo
Snack here https://snack.expo.io/@ecatch/test-sqlite

**Comment by martin-ecatch** [comment]
Oh, the plot thickens :-). I hadn't tried the sample code on Android. I saw the escape/unescape code and assumed that it was solved for Android. Also, I haven't ran into issues with PouchDB indexes the way I did for iOS.
But you are right, the sample code totally breaks on Android, too. How...?
PS: typing "\u0000" to the input string would work fine, because it would be the 6 characters "\", "u" and four times "0". It's not possible (AFAIK) to actually type \u0000 character.

**Comment by lukmccall** [comment]
I've tried to dig into this issue and tracked where this `null character` disappeared.
Unfortunately,  It seems to happen somewhere in the bridge (in c land 😢).
Moreover, when I've tried to debug it I got the `null character` correctly inside the java module, so SQL query gets and returns the correct values, but when we send SQL data from java to js, `null character` and text after disappears. I suppose that happens because in c string is null-terminated.
Pretty weird is that we can send "\u0000" from js to java without any problem.

I've tried to debug this only on Android, but I'm thinking iOS is the same story.
